src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -263,10 +264,13 @@ public CompiledScript compileInternal(Script script) {
             return compiled;
         }
 
+        // Name will remain null for inline scripts.
```

Comment:
I skipped file scripts here, as if it was a file script it had already been processed, but added a note to the comment to clarify.